We need to quit plymouth before running firstboot (#679171)

Hiding the splash is not enough. If plymouth is running
when we execute firstboot, pressing the Enter key causes
it to exit.
---
 init/firstboot                      |    4 ----
 systemd/firstboot-graphical.service |    2 +-
 2 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/init/firstboot b/init/firstboot
index 6e92a84..7286b15 100755
--- a/init/firstboot
+++ b/init/firstboot
@@ -65,13 +65,9 @@ case "$1" in
 
         . /etc/profile.d/lang.sh
 
-        /usr/bin/plymouth --hide-splash
-
         /usr/sbin/firstboot $args
         RETVAL=$?
 
-        /usr/bin/plymouth --show-splash
-
         # If firstboot succeeded, chkconfig it off so we don't see the message
         # every time about starting up firstboot.
         if [ "$RETVAL" -eq 0 ]; then
diff --git a/systemd/firstboot-graphical.service b/systemd/firstboot-graphical.service
index fb18796..f4d6934 100644
--- a/systemd/firstboot-graphical.service
+++ b/systemd/firstboot-graphical.service
@@ -6,12 +6,12 @@ Conflicts=firstboot-text.service
 
 [Service]
 Environment=RUNLEVEL=5
+ExecStartPre=-/bin/plymouth quit
 ExecStart=/etc/init.d/firstboot start
 TimeoutSec=0
 StandardInput=tty
 RemainAfterExit=yes
 Type=oneshot
-SysVStartPriority=99
 
 [Install]
 WantedBy=graphical.target
-- 
1.7.3.2

